samples/MicStream.js

```diff
@@ -0,0 +1,378 @@
+  const record = require('node-record-lpcm16');
+  //https://www.npmjs.com/package/node-record-lpcm16
+
+  // Imports the Google Cloud client library
+  const speech = require('@google-cloud/speech');
+
+  // Creates a client
+  const client = new speech.SpeechClient();
+
+  //Creates an array of interim results, for testing
+  var InterimResults = [];
+
+  //defensive code in case user doesn't specify audio file type
+  if (filename.substr(filename.length - 4) !== `.wav`)
+    filename = filename + `.wav`;
+
+  var file = fs.createWriteStream(filename, {
+    encoding: 'binary',
+  });
+
+  // Create a recognize stream
+  const recognizeStream = client
+    .streamingRecognize(request)
+    .on('error', console.error)
+    .on('data', data => {
+      if (
+        data.results[0] &&
+        data.results[0].alternatives[0] &&
+        data.results[0].isFinal
+      )
+        process.stdout.write(
+          `micStreamRecognize: Transcript; 
+          ${data.results[0].alternatives[0].transcript} \n`
+        );
+      streamtranscription = data.results[0].alternatives[0].transcript;
+      var a = [];
+      a.push(data.results[0].alternatives[0].transcript);
+      a.push(data.results[0].isFinal.toString());
+      if (!data.results[0].isFinal)
+        a.push(data.results[0].stability.toString());
+      InterimResults.push(a); //CF: array of alternatives for testing
+    });
+
+  // Start recording and send the microphone input to the Speech API
+  record
+    .start({
+      sampleRateHertz: sampleRateHertz,
+      threshold: 0, //silence threshold
+      // Other options, see https://www.npmjs.com/package/node-record-lpcm16#options
+      //verbose: true, //CF: verbose true just outputs 1/2 the samplerateHertz every 1/2 sec.  Not all that necessary.
+      recordProgram: 'rec', // was 'rec' : Try also "arecord" or "sox"
+      silence: '5.0', //seconds of silence before ending
+      //device: null
+    })
+    .on(`data`, chunk => {
+      file.write(chunk);
+    })
+    .on('error', console.error)
+    .pipe(recognizeStream);
+
+  console.log(
+    `micStreamRecognize: Listening for ${secondsofRecording} seconds, press Ctrl+C to stop.`
+  );
+
+  //CF: records for secondsofRecording seconds.
+  setTimeout(function() {
+    record.stop();
+    console.log(`micStreamRecognize: Recording complete!`);
+    console.log(
+      `micStreamRecognize: There are ${
+        InterimResults.length
+      } InterimResults generated from the recording`
+    );
+    console.log(
+      `**Note that best guess/ non-interim results below are included below, true, and don't include confidence value**`
+    );
+
+    for (var i in InterimResults) {
+      console.log(`  `, i, `)`, InterimResults[i].toString());
+    }
+    eventEmitter.emit(`micStreamRecognize: complete`);
+  }, secondsofRecording * 1000);
+  return InterimResults;
+  // [END micStreamRecognize]
+}
+
+function wavFileToText(filename, encoding, sampleRateHertz, languageCode) {
```

Comment:
We already have a samples for this:
 - https://cloud.google.com/speech-to-text/docs/sync-recognize#speech-sync-recognize-nodejs
 - https://github.com/googleapis/nodejs-speech/blob/master/samples/recognize.js#L27-L70

Remove from the mic streaming sample code